zerver/lib/narrow.py

```diff
@@ -44,7 +44,7 @@ def narrow_filter(event: Mapping[str, Any]) -> bool:
                 if operand.lower() != message["sender_email"].lower():
                     return False
             elif operator == "is" and operand == "private":
-                if message["type"] != "private":
+                if "is_private" not in flags:
```

Comment:
This line doesn't need to change.  What we're doing here is denormalizing the table (i.e. duplicating the data), and checking the always-present source (`message["type"]` is the correct method).

Reply:
(And this implies we don't need to change tests)